File: Lv2/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int ptr = 0, day, count;
    
    for (day = 0; ; day++) {
        if (ptr >= progresses.size()) break;
        
        for (int i = 0; i < progresses.size(); i++) 
            progresses.at(i) += speeds.at(i);
        
        for (count = 0;;count++) {
            if (ptr < progresses.size() && progresses.at(ptr) >= 100) {
                ptr++;
                continue;
            }
            else break;
        }
            
        if (count != 0) answer.push_back(count);
    }
    
    
    
    return answer;
}

File: Lv2/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    count = 1
    days = [math.ceil((100 - p) / s) for p, s in zip(progresses, speeds)]
    
    for i in range(len(days) - 1):
        if days[i] < days[i + 1]:
            answer.append(count)
            count = 1
        else:
            days[i + 1] = days[i]
            count += 1
    
    answer.append(count)
    
    return answer
